orchagent/muxorch.cpp

```diff
@@ -795,22 +804,48 @@ bool MuxNbrHandler::enable(bool update_rt)
         IpPrefix pfx = it->first.to_string();
         if (update_rt)
         {
-            if (remove_route(pfx) != SAI_STATUS_SUCCESS)
-            {
-                return false;
-            }
-            updateTunnelRoute(nh_key, false);
+            bulk_route_ctx_list.push_back(MuxRouteBulkContext(pfx, false));
         }
 
         it++;
     }
 
+    if (update_rt)
+    {
+        if (!createBulkRouteEntries(bulk_route_ctx_list))
+        {
+            gRouteBulker.clear();
+            return false;
+        }
+        gRouteBulker.flush();
+        if (!processBulkRouteEntries(bulk_route_ctx_list))
+        {
+            gRouteBulker.clear();
+            return false;
+        }
+
+        it = neighbors_.begin();
+        while (it != neighbors_.end())
```

Comment:
We've introduced three loops now in the new code. Can we optimize it?

Reply:
I took the loop out from the comment below, I believe the two left over are necessary, we need the neighbors enabled before we update routes to point to their next hops